### ISSUE TITLE
UserByRoleNameIndex table has invalid data

### ISSUE THREAD

**Issue by intellimedhu** [OrchardCMS/OrchardCore#14613]
I use UserByRoleNameIndex in yessql queries to fetch users in a specific roles. The queries suddenly returned empty results, and upon checking, I discovered that this is because the table contains the following data:
![image](https://github.com/OrchardCMS/OrchardCore/assets/65296389/9291a69f-f79c-4f63-94b6-08b3d9fd601a)

Why are roles missing? Why are there negative values in the 'Count' column? The main question is, how can this be fixed immediately?

**Comment by intellimedhu** [comment]
I further examined the situation. When I look up a user in the dashboard, it appears that their permissions are in order. However, even when I save on the user, the correct data still doesn't get added to the reduced index.

**Comment by hishamco** [comment]
This could happen due to the following lines:

(https://github.com/OrchardCMS/OrchardCore/blob/1c6612678a8b67f33a48b740406b8d10a3a23535/src/OrchardCore/OrchardCore.Users.Core/Indexes/UserByRoleNameIndex.cs#L54-L57)

**Comment by MikeAlhayek** [comment]
Yes I think the issue is what @hishamco pointed out. I think line 55 should be 
```
index.Count = map.Sum(x => x.Count); 
```

**Comment by sebastienros** [comment]
Can anyone else check their table to find other occurrences of this issue and verify if the counts are correct?

> how can this be fixed immediately?

@intellimedhu you can update the column with the expected counts by checking which users have these roles (using a SQL query).

What version are you in, and would you be able to reproduce the issue from a new site?

**Comment by MikeAlhayek** [comment]
@intellimedhu I can't reproduce the issue.
